build-calico-resource.sh

```diff
@@ -50,29 +52,47 @@ for arch in ${arches}; do
   rm -rf resource-build-$arch
   mkdir resource-build-$arch
   pushd resource-build-$arch
-
-  if [ $arch = "amd64" ]; then
-    fetch_and_validate \
-      https://github.com/projectcalico/calicoctl/releases/download/v1.6.5/calicoctl
-    fetch_and_validate \
-      https://github.com/projectcalico/cni-plugin/releases/download/v1.11.8/calico
-    fetch_and_validate \
-      https://github.com/projectcalico/cni-plugin/releases/download/v1.11.8/calico-ipam
-  elif [ $arch = "arm64" ]; then
-    fetch_and_validate \
-      https://people.canonical.com/~kwmonroe/calico-2.6-arm64/calicoctl
-    fetch_and_validate \
-      https://people.canonical.com/~kwmonroe/calico-2.6-arm64/calico
-    fetch_and_validate \
-      https://people.canonical.com/~kwmonroe/calico-2.6-arm64/calico-ipam
-  else
-    echo "$0: Unsupported architecture: $arch"
-    exit 1
-  fi
+  fetch_and_validate \
+    https://github.com/projectcalico/calicoctl/releases/download/$calicoctl_version/calicoctl-linux-$arch
+  fetch_and_validate \
+    https://github.com/projectcalico/cni-plugin/releases/download/$calico_cni_version/calico-$arch
+  fetch_and_validate \
+    https://github.com/projectcalico/cni-plugin/releases/download/$calico_cni_version/calico-ipam-$arch
+  mv calicoctl-linux-$arch calicoctl
+  mv calico-$arch calico
+  mv calico-ipam-$arch calico-ipam
 
   chmod +x calicoctl calico calico-ipam
   tar -zcvf ../calico-$arch.tar.gz .
 
   popd
   rm -rf resource-build-$arch
 done
+
+# calico-upgrade resource
+for arch in ${arches}; do
+  rm -rf resource-build-upgrade
+  mkdir resource-build-upgrade
+  pushd resource-build-upgrade
```

Comment:
curious why pushd/popd vs cd/cd ..

Reply:
They were used elsewhere in the file. I used them for consistency.

I personally prefer this anyway:
```
(cd
  do stuff
)
```